src/ngx_http_graphite_module.c

```diff
@@ -1813,8 +1929,14 @@ ngx_http_graphite_timer_event_handler(ngx_event_t *ev) {
 
             if (nl > part) {
 
-                if (sendto(fd, part, nl - part + 1, 0, (struct sockaddr*)&sin, sizeof(sin)) == -1)
-                    ngx_log_error(NGX_LOG_ALERT, ev->log, 0, "graphite can't send udp packet");
+                if (ngx_strcmp(gmcf->protocol.data, "udp") == 0) {
+                    if (sendto(fd, part, nl - part + 1, 0, (struct sockaddr*)&sin, sizeof(sin)) == -1)
+                        ngx_log_error(NGX_LOG_ALERT, ev->log, 0, "graphite can't send udp packet");
+                }
+                if (ngx_strcmp(gmcf->protocol.data, "tcp") == 0) {
+                    if (send(fd, part, nl - part + 1, 0) < 0)
```

Comment:
This operation may fail with EWOULDBLOCK when the buffer is full. You can either handle the error, or check if the file descriptor is ready for output before writing to it.